didcomm/pack_encrypted.py

```diff
@@ -87,12 +93,51 @@ async def pack_encrypted(
     :return: A pack result consisting of a packed message as a JSON string
              and an optional service metadata with an endpoint to be used to transport the packed message.
     """
+    if message.to is not None and get_did(to) not in message.to:
+        raise ValueError()
+
+    if frm is not None and message.frm is not None and get_did(frm) != message.frm:
+        raise ValueError()
+
+    if pack_config is None:
+        pack_config = PackEncryptedConfig()
+
+    if pack_params is None:
+        pack_params = PackEncryptedParameters()
+
+    resolvers_config = get_effective_resolvers(resolvers_config)
+
+    msg = to_bytes(json_dumps(message.as_dict()))
+
+    to_kids = []
+    from_kid = None
+    sign_from_kid = None
+
+    if sign_frm is not None:
+        sign_result = await sign(msg, sign_frm, resolvers_config)
+
+        msg = sign_result.msg
+        sign_from_kid = sign_result.sign_frm_kid
+
+    if frm is not None:
+        authcrypt_result = await authcrypt(msg, to, frm, pack_config.enc_alg_auth, resolvers_config)
+
+        msg = authcrypt_result.msg
+        to_kids = authcrypt_result.to_kids
+        from_kid = authcrypt_result.from_kid
+
+    if frm is None or pack_config.protect_sender_id:
+        anoncrypt_result = await anoncrypt(msg, to, pack_config.enc_alg_anon, resolvers_config)
```

Comment:
We need to use exactly the same `to` in case of `protect_sender_id` as in authcrypt